dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/datafile/file/FileMetadataServiceTest.java

```diff
@@ -0,0 +1,126 @@
+package edu.harvard.iq.dataverse.datafile.file;
+
+import edu.harvard.iq.dataverse.DataverseRequestServiceBean;
+import edu.harvard.iq.dataverse.EjbDataverseEngine;
+import edu.harvard.iq.dataverse.engine.command.impl.DeleteProvJsonCommand;
+import edu.harvard.iq.dataverse.engine.command.impl.PersistProvJsonCommand;
+import edu.harvard.iq.dataverse.persistence.datafile.DataFile;
+import edu.harvard.iq.dataverse.persistence.datafile.FileMetadata;
+import edu.harvard.iq.dataverse.provenance.UpdatesEntry;
+import org.junit.Assert;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.HashMap;
+import java.util.Set;
+
+import static io.vavr.collection.HashMap.of;
+
+@ExtendWith(MockitoExtension.class)
+class FileMetadataServiceTest {
+
+    @InjectMocks
+    private FileMetadataService fileMetadataService;
+
+    @Mock
+    private EjbDataverseEngine commandEngine;
+
+    @Mock
+    private DataverseRequestServiceBean dvRequestService;
+
+    @Test
+    public void updateFileMetadataWithProvFreeForm() {
+        //given
+        String checksum = "testChecksum";
+        String provFree = "provFree";
+
+        DataFile dataFile = new DataFile();
+        dataFile.setChecksumValue(checksum);
+        FileMetadata fileMetadata = new FileMetadata();
+        fileMetadata.setDataFile(dataFile);
+
+        HashMap<String, UpdatesEntry> provenanceUpdates = of(checksum, new UpdatesEntry(dataFile, "prov", false, provFree))
+                .toJavaMap();
+
+        //when
+        FileMetadata updatedFile = fileMetadataService.updateFileMetadataWithProvFreeForm(fileMetadata, provenanceUpdates);
+
+        //then
+        Assert.assertEquals(provFree, updatedFile.getProvFreeForm());
+    }
+
+    @SuppressWarnings("SimplifiableJUnitAssertion")
+    @Test
+    public void updateFileMetadataWithProvFreeForm_WithNoProvenance() {
+        //given
+        String checksum = "testChecksum";
+
+        DataFile dataFile = new DataFile();
+        dataFile.setChecksumValue(checksum);
+        FileMetadata fileMetadata = new FileMetadata();
+        fileMetadata.setDataFile(dataFile);
+
+        //when
+        fileMetadataService.updateFileMetadataWithProvFreeForm(fileMetadata, new HashMap<>());
+
+        //then
+        Assert.assertEquals(null, fileMetadata.getProvFreeForm());
```

Comment:
You can do `assertNull`

Reply:
I discussed it with Darek earlier, and the conclusion was that usually Equals is better since when error occurs it prints expected and the actual result. When ex. AssertTrue is used it onl;y prints that error there occurred and not the actual values.

Reply:
code: `assertNull("djksdjsk");`
message: `java.lang.AssertionError: expected null, but was:<djksdjsk>`
I guess that you had to add warning suppression for this case?

When you assert something for true and it fails then its quite easy to conduct what was the actual value?
`assertTrue` have better type type checking than `assertEquals` (compilation error vs runtime error)

Reply:
Ah, yeah then assertNull is allright.

The argument about assertTrue was that you won't be able to know from command line quickly what were the values. Here is good example 

```
From assertEquals(a, b)
Error: Expected "Hi"; was "Hello"

From assertTrue:
Error: Expected true; was false
```